src/Servers/Kestrel/Core/src/Internal/Infrastructure/KestrelTrace.cs

```diff
@@ -151,233 +151,241 @@ internal class KestrelTrace : IKestrelTrace
         {
-            _connectionPause(_logger, connectionId, null);
+            _connectionPause(_connectionsLogger, connectionId, null);
         }
 
         public virtual void ConnectionResume(string connectionId)
         {
-            _connectionResume(_logger, connectionId, null);
+            _connectionResume(_connectionsLogger, connectionId, null);
         }
 
         public virtual void ConnectionKeepAlive(string connectionId)
         {
-            _connectionKeepAlive(_logger, connectionId, null);
+            _connectionKeepAlive(_connectionsLogger, connectionId, null);
         }
 
         public virtual void ConnectionRejected(string connectionId)
         {
-            _connectionRejected(_logger, connectionId, null);
+            _connectionRejected(_connectionsLogger, connectionId, null);
         }
 
         public virtual void ConnectionDisconnect(string connectionId)
         {
-            _connectionDisconnect(_logger, connectionId, null);
+            _connectionDisconnect(_connectionsLogger, connectionId, null);
         }
 
         public virtual void ApplicationError(string connectionId, string traceIdentifier, Exception ex)
         {
-            _applicationError(_logger, connectionId, traceIdentifier, ex);
+            _applicationError(_generalLogger, connectionId, traceIdentifier, ex);
         }
 
         public virtual void ConnectionHeadResponseBodyWrite(string connectionId, long count)
         {
-            _connectionHeadResponseBodyWrite(_logger, connectionId, count, null);
+            _connectionHeadResponseBodyWrite(_generalLogger, connectionId, count, null);
         }
 
         public virtual void NotAllConnectionsClosedGracefully()
         {
-            _notAllConnectionsClosedGracefully(_logger, null);
+            _notAllConnectionsClosedGracefully(_connectionsLogger, null);
         }
 
         public virtual void ConnectionBadRequest(string connectionId, Microsoft.AspNetCore.Http.BadHttpRequestException ex)
         {
-            _connectionBadRequest(_logger, connectionId, ex.Message, ex);
+            _connectionBadRequest(_badRequestsLogger, connectionId, ex.Message, ex);
         }
 
         public virtual void RequestProcessingError(string connectionId, Exception ex)
         {
-            _requestProcessingError(_logger, connectionId, ex);
+            _requestProcessingError(_badRequestsLogger, connectionId, ex);
         }
 
         public virtual void NotAllConnectionsAborted()
         {
-            _notAllConnectionsAborted(_logger, null);
+            _notAllConnectionsAborted(_connectionsLogger, null);
         }
 
         public virtual void HeartbeatSlow(TimeSpan heartbeatDuration, TimeSpan interval, DateTimeOffset now)
         {
-            _heartbeatSlow(_logger, now, heartbeatDuration, interval, null);
+            _heartbeatSlow(_connectionsLogger, now, heartbeatDuration, interval, null);
```

Comment:
Nit: While the heartbeat does loop over connections, this log is usually an indicator of threadpool starvation.

```suggestion
            _heartbeatSlow(_generalLogger, now, heartbeatDuration, interval, null);
```

Reply:
Applied